dockstore-webservice/src/main/java/io/dockstore/webservice/core/metrics/MetricsDataS3Client.java

```diff
@@ -0,0 +1,166 @@
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+import software.amazon.awssdk.services.s3.model.HeadObjectRequest;
+import software.amazon.awssdk.services.s3.model.ListObjectsV2Request;
+import software.amazon.awssdk.services.s3.model.ListObjectsV2Response;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.awssdk.services.s3.model.S3Object;
+
+public class MetricsDataS3Client {
+    private static final Logger LOG = LoggerFactory.getLogger(MetricsDataS3Client.class);
+    private final S3Client s3;
+    private final String bucketName;
+
+    public MetricsDataS3Client(String bucketName) throws URISyntaxException {
+        this.bucketName = bucketName;
+        //TODO should not need to hardcode region since buckets are global, but http://opensourceforgeeks.blogspot.com/2018/07/how-to-fix-unable-to-find-region-via.html
+        this.s3 = S3Client.builder().region(Region.US_EAST_1).build();
+    }
+
+    /**
+     * This constructor is only used for testing with the localstack endpoint
+     * @param bucketName
+     * @param endpointOverride
+     */
+    public MetricsDataS3Client(String bucketName, URI endpointOverride) {
+        this.bucketName = bucketName; // TODO:
+        //TODO should not need to hardcode region since buckets are global, but http://opensourceforgeeks.blogspot.com/2018/07/how-to-fix-unable-to-find-region-via.html
+        this.s3 = S3Client.builder().region(Region.US_EAST_1).endpointOverride(endpointOverride).build();
+    }
+
+    /**
+     * Creates an S3 object containing metrics data for the given GA4GH tool and tool version
+     *
+     * @param toolId The GA4GH Tool ID
+     * @param versionName The GA4GH ToolVersion name
+     * @param platform The platform that the metrics data is from
+     * @param fileName The file name to use. Should be the time that the data was submitted in milliseconds since epoch appended with '.json'
+     * @param ownerUserId The user id of the owner (user that sent the metrics data)
+     * @param metricsData The metrics data in JSON format
+     */
+    public void createS3Object(String toolId, String versionName, String platform, String fileName, long ownerUserId, String metricsData) {
+        try {
+            String key = generateKey(toolId, versionName, platform, fileName);
+            Map<String, String> metadata = Map.of(ObjectMetadata.TOOL_ID.toString(), toolId,
+                    ObjectMetadata.VERSION_NAME.toString(), versionName,
+                    ObjectMetadata.PLATFORM.toString(), platform,
+                    ObjectMetadata.FILENAME.toString(), fileName,
+                    ObjectMetadata.OWNER.toString(), String.valueOf(ownerUserId));
+            PutObjectRequest request = PutObjectRequest.builder()
+                    .bucket(bucketName)
+                    .key(key)
+                    .metadata(metadata)
+                    .contentType(MediaType.APPLICATION_JSON)
```

Comment:
To put it into S3, it doesn't matter because I don't think the S3 client checks. For processing the data into aggregated metrics on Dockstore, we will want `metricsData` to be JSON so we can extract the information